provider/cmd/pulumi-resource-cloudinit/schema.json

```diff
@@ -85,161 +93,173 @@
                 },
                 "gzip": {
-                    "type": "boolean"
+                    "type": "boolean",
+                    "description": "Specify whether or not to gzip the `rendered` output. Defaults to `true`.\n"
                 },
                 "parts": {
                     "type": "array",
                     "items": {
                         "$ref": "#/types/cloudinit:index/ConfigPart:ConfigPart"
-                    }
+                    },
+                    "description": "A nested block type which adds a file to the generated cloud-init configuration. Use multiple `part` blocks to specify multiple files, which will be included in order of declaration in the final MIME document.\n"
                 },
                 "rendered": {
                     "type": "string",
-                    "description": "rendered cloudinit configuration\n"
+                    "description": "The final rendered multi-part cloud-init config.\n"
                 }
             },
             "required": [
-                "parts",
+                "base64Encode",
+                "boundary",
+                "gzip",
                 "rendered"
             ],
             "inputProperties": {
                 "base64Encode": {
                     "type": "boolean",
-                    "willReplaceOnChanges": true
+                    "description": "Specify whether or not to base64 encode the `rendered` output. Defaults to `true`, and cannot be disabled if gzip is `true`.\n"
                 },
                 "boundary": {
                     "type": "string",
-                    "willReplaceOnChanges": true
+                    "description": "Specify the Writer's default boundary separator. Defaults to `MIMEBOUNDARY`.\n"
                 },
                 "gzip": {
                     "type": "boolean",
-                    "willReplaceOnChanges": true
+                    "description": "Specify whether or not to gzip the `rendered` output. Defaults to `true`.\n"
                 },
                 "parts": {
                     "type": "array",
                     "items": {
                         "$ref": "#/types/cloudinit:index/ConfigPart:ConfigPart"
                     },
-                    "willReplaceOnChanges": true
+                    "description": "A nested block type which adds a file to the generated cloud-init configuration. Use multiple `part` blocks to specify multiple files, which will be included in order of declaration in the final MIME document.\n"
                 }
             },
-            "requiredInputs": [
-                "parts"
-            ],
             "stateInputs": {
                 "description": "Input properties used for looking up and filtering Config resources.\n",
                 "properties": {
                     "base64Encode": {
                         "type": "boolean",
-                        "willReplaceOnChanges": true
```

Comment:
This has been removed from quite a few resources, is this a concern?

Reply:
I don't think so, but I don't have a lot of context here. This is also something users can control with our own `replaceOnChanges` option if it becomes an issue.